providers/dns/cloudflare/cloudflare.go

```diff
@@ -57,7 +65,8 @@ func NewDNSProvider() (*DNSProvider, error) {
 	if err != nil {
 		var errT error
 		values, errT = env.GetWithFallback(
-			[]string{"CLOUDFLARE_API_TOKEN", "CF_API_TOKEN"},
+			[]string{"CLOUDFLARE_DNS_API_TOKEN", "CF_DNS_API_TOKEN"},
+			[]string{"CLOUDFLARE_ZONE_API_TOKEN", "CF_ZONE_API_TOKEN", "CLOUDFLARE_DNS_API_TOKEN", "CF_DNS_API_TOKEN"},
```

Comment:
```suggestion
			[]string{"CLOUDFLARE_ZONE_API_TOKEN", "CF_ZONE_API_TOKEN"},
			[]string{"CLOUDFLARE_DNS_API_TOKEN", "CF_DNS_API_TOKEN"},
```

Reply:
Hm, this makes `CLOUDFLARE_ZONE_API_TOKEN` mandatory, doesn't it?

Reply:
yes but your approach will not work.